### DISCUSSION TITLE
Optimise Tournament Bracket file sizes / Remove unneeded values from json objects for tournaments.

### DISCUSSION THREAD

**Discussion by Coppertine** [ppy/osu#18681] [Ideas] [answered]
This has become a bit of an issue, espcially with Round of 128 tournaments where you would replicate the bracket to share across to multiple streamers for the lazer client.
However, due to the bloat with each user and beatmap object, the case of 128 teams (each including 1 player) comes to 11mb of storage for the json file itself.
This itself does not include beatmaps which can easily become 20-30mb.

On smaller scales, a Round of 32 tournament (single elimination) can go upwards of 7-8mb towards grand finals, making sharing across discord (to share to other streamers) a bit more cumbersome to handle having to zip up a json file.

The main issue with the storage of the bracket.json file is the bloat of the APIUser and APIBeatmap values where every single value is included and not just the Title, ID, Difficulty, Image, any anything else used for the tournament client.

**Comment by konotorii** [comment]
Would add onto this by saying that these extra APIUser and APIBeatmap values could be stored separately to still keep these values and still reducing the file size of the bracket file.

**Comment by peppy** [comment]
I'm not against this being done. It shouldn't be a very hard one to improve if someone feels like stepping up to make the changes.

**Comment by frenzibyte** [comment, marked as answer]
Should be improved magnificently starting with https://github.com/ppy/osu/pull/18750.

**Reply by frenzibyte** [reply]
Along with https://github.com/ppy/osu/pull/18751.

**Reply by Coppertine** [reply]
Thank you for your work!
